components/script/dom/request.rs

```diff
@@ -239,11 +232,7 @@ impl Request {
         }
 
         // Step 19
-        let credentials = init
-            .credentials
-            .as_ref()
-            .map(|m| m.clone().into())
-            .or(fallback_credentials);
+        let credentials = init.credentials.as_ref().map(|m| m.clone().into());
```

Comment:
Suggestion: this can be combined with the subsequent if statement i.e
```
if let Some(init_credentials) = init.credentials.as_ref() {
    let credentials = init_credentials.clone().into();
    request.credentials_mode = credentials;
}
```

It is verbose, but seems to be consistent with the style of the surrounding code for other RequestInit options.

Reply:
Thanks for pointing this out. It looks alot more natural. Just added in fix-up commit